source/ID/concordium-did.rst

```diff
@@ -0,0 +1,479 @@
+               "w" / "x" / "y" / "z"
+
+.. note::
+    The network part of a CCD DID is optional.
+    *If omitted the network is assumed to be "mainnet" by default*.
+
+.. note::
+    The ``subindex`` part of the smart contract instance ``sci`` is optional.
+    *If omitted the subindex is assumed to be `0` by default.*
+
+Smart Contract Instance Reference Syntax
+----------------------------------------
+
+A DID reference is a DID with additional data, such as query parameters.
+Concordium smart contract entrypoints can be addressed using DID references.
+
+.. code-block:: ABNF
+
+  ccd-sc-ref = "/" entrypoint *1("?" param-name "=" 1*(base58char))
+  param-name = "parameter_json" / "parameter_binary"
+  entrypoint = 1*(ALPHA / DIGIT / punctuation)
+  punctuation = "!" / DQUOTE / "#" / "$" / "%" / "&" / "'" / "(" /
+                ")" / "*" / "+" / "," / "-" / "." / "/" / ";" /
+                "<" / "=" / ">" / "?" / "@" / "[" / "\" / "]" /
+                "^" / "_" / "`" / "{" / "|" / "}" / "~"
+
+See the details about *dereferencing* a smart contract instance reference in :ref:`sci-reference`.
+
+Examples
+--------
+
+An account on testnet:
+
+``did:ccd:testnet:acc:3ZFGxLtnUUSJGW2WqjMh1DDjxyq5rnytCwkSqxFTpsWSFdQnNn``
+
+A smart contract instance on the default network (``mainnet``):
+
+``did:ccd:sci:12:0``
+
+A public key related to ``mainnet``:
+
+``did:ccd:mainnet:pkc:0c7f4421e44a4385850b883e3bbf098f5a9853ef6f1a862c2ce2856381b5f5e3``
+
+A smart contract instance with the ``getKeys`` entrypoint that does not take any parameters
+
+``did:ccd:sci:321/getKeys``
+
+A smart contract instance with the ``getCredentials`` entrypoint taking a parameter
+
+``did:ccd:sci:123/getCredentials?parameter=QWWfHFtjPSLJdrz1beyVPzFmgS2ECbJVe1k2LXPwUGMZRjjpL``
+
+
+Concordium DID Documents
+========================
+
+.. TODO add formal DID documents
+
+Account DID
+-----------
+
+The Account DID Document MUST contain the following data:
+
+- ``@context``
+- ``id`` - the DID of the account.
+- ``accountCredentials`` - a list of account holders represented by their account credentials ``credential-i`` for ``i = 1..M``, where ``M`` is the number of credentials.
+  The credentials contain several (at least one) signature verification keys ``key-i-j`` for ``j = 1..N_i``, where ``N_i`` is the number of keys, which can be different for each credential.
+  For each credential:
+  - ``verificationMethod`` - uses a threshold verification scheme that specifies public keys ``key-i-j`` and a signature threshold ``R_j``.
+- ``verificationMethod`` - account's verification method; it is again a threshold scheme requiring at least ``T`` credentials to sign.
+
+The document MAY include any other public data of a Concordium account.
+
+.. code-block:: json
+
+  {
+    "@context": [
+      "https://www.w3.org/ns/did/v1",
+      "Concordium DID URI"
+    ],
+    "id": "did:ccd:NET:acc:ADDR",
+    "accountCredentials": [
+      {
+        "verificationMethod": [
```

Comment:
Would be nice to have the formal definition of "verificationMethod" (i.e., defining its properties e.g., id, controller, type, etc)

Reply:
I added a reference to the w3c did standard